source/exec/docker.d

```diff
@@ -95,9 +99,8 @@ class Docker: IExecProvider
 			return typeof(return)("Maximum number of parallel compiles has been exceeded. Try again later.", false);
 		}
 
-		++queueSize_;
-		scope(exit)
-			--queueSize_;
+		synchronized { ++queueSize_; }
+		scope(exit) synchronized { --queueSize_; }
```

Comment:
This is not correct since the two `synchronized` blocks use different mutexes and will essentially race with each other.

The biggest problem however is that the synchronization is thread-wide meaning that it will stop the whole eventloop, essentially serving one user at a time. You need to use vibe.d specific synchronization primitives here.

I suggest you move the synchronization in the parallel section of the constructor.